middleware/cacheHandler.js

```diff
@@ -0,0 +1,61 @@
+const config = require('config')
+const urlGetter = require('url')
+
+const getUrl = (req) => {
+  return urlGetter.format({
+    protocol: req.protocol,
+    host: req.get('host'),
+    pathname: req.originalUrl
+  })
+}
+
+class CachedData {
+  constructor (requestUrl, timeStamp, data) {
+    this.url = requestUrl
+    this.timeStamp = timeStamp
+    this.data = data
+  }
+
+  getData () {
+    return this.data
+  }
+
+  isExpired () {
+    return this.timeStamp + cacheMillisecondsToLive < Date.now()
+  }
+}
+
+const cacheDataMap = new Map()
+const cacheMinutesToLive = 10
+const cacheMillisecondsToLive = cacheMinutesToLive * 60 * 1000
+
+const saveToCache = (url, data) => {
+  cacheDataMap.set(url, new CachedData(url, Date.now(), data))
+}
+
+const isInCache = (requestedUrl) => {
+  if (requestedUrl) {
+    return !requestedUrl.isExpired()
+  }
+  return null
+}
+const handleCachedData = (req, res, next) => {
+  if (config.isCacheEnabled) {
+    const requestedUrl = cacheDataMap.get(getUrl(req))
```

Comment:
What if it's a post request with the same url as a get request but a different http method? the check of :
```
if (req.method === 'GET')
```
should be at the first line of the middleware, and if the method is not get then you should call next